iceoryx_posh/test/moduletests/test_popo_smart_chunk_common.hpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_TEST_POPO_SMART_CHUNK_COMMON_HPP
+#define IOX_TEST_POPO_SMART_CHUNK_COMMON_HPP
+
+#include "iceoryx_posh/internal/popo/rpc_interface.hpp"
+#include "iceoryx_posh/popo/publisher.hpp"
```

Comment:
Do we really need this include?

Reply:
yes, it contains the `PublisherInterface`

Reply:
That should be in separate file. If it's not too much of a hassle, I would appreciate if you do it on this PR.

Reply:
@mossmaurice you are right but there is more in that file that needs to be split up, e.g. move the `impl` to a separate file like with client and server. Since this would affect the typed and untyped publisher and subscriber I would postpone it since it does not belong to this PR. I'm happy to create an issue, though.

Reply:
@mossmaurice done with #1173